milestones/3_human_brain_mapping_journal_submission/experiments/ex_2_pli_generation_optimization/example_pli_analysis.m

```diff
@@ -0,0 +1,214 @@
+    T = 100/(2*B);                % ~100 effective points
+    N = round(T*fd/2)*2;
+    K = fix((m-N/2-cut*2)/(N/2)); % number of windows, 50% overlap
+    V = nchoosek(R,2);            % number of ROI pairs
+
+    %% Iterate over each window and calculate pairwise corrected aec
+    result = struct();
+    pli = zeros(NUM_REGIONS, NUM_REGIONS, K);
+
+    % Boolean mask to convert to a square matrix
+    ind = logical(triu(ones(R),1));
+
+    % Loop over time windows
+    parfor k = 1:K
+        fprintf("Calculating window '%d' \n", k);
+        ibeg = (N/2)*(k-1) + 1;
+        iwind = ibeg:ibeg+N-1;
+
+        % Calculate PLI
+        theta_win = theta(iwind, :);
+        pli_temp =  calculate_pli(theta_win, ind, V);
+
+        % Correct it using surrogate analysis
+        pli_temp_corr = surrogate_analysis(theta, ind, V, pli_temp);
+        pli(:, :, k) = pli_temp_corr;
+    end
+
+    result.wpli = pli;
+
+    % Bundling some metadata that could be useful along with the graph
+    result.window_size = window_size;
+    result.step_size = step_size;
+    result.labels = LABELS;
+
+    % Save the result structure at the right spot
+    save(participant_out_path, 'result');
+
+end
+
+function [pli_temp] = calculate_pli(theta_win, ind, V)
+%% CALCULATE PLI helper function to wrap the calculation of PLI
+
+        [N,R] = size(theta_win);
+        pli_vector = zeros(V,1);
+        
+        % loop over all possible ROI pairs
+        for jj = 2:R
+            ii = 1:jj-1;
+            indv = ii + sum(1:jj-2);
+            % Phase difference
+            RP = bsxfun(@minus,theta_win(:, jj),theta_win(:, ii));
+            srp = sin(RP);
+            pli_vector(indv) = abs(sum(sign(srp),1))/N;
+        end
+
+        % Attempt at converting from a vector to a pli matrix
+        pli_temp = zeros(R, R);
+        pli_temp(ind) = pli_vector(:);
+        pli_temp = pli_temp + pli_temp';
+end
+
+function pli_temp = calculate_pli_surrogate(theta, ind, V)
+    % Given a multivariate data, returns phase lag index matrix
+    % Modified the mfile of 'phase synchronization'
+    
+    [N,R] = size(theta);
+    num_pts = length(theta);
+    splice = randi(num_pts);  % determines random place in signal where it will be spliced
+
+    a_sig_splice = [theta(splice:num_pts,:); theta(1:splice-1,:)];  % %This is the randomized signal
+    
+    pli_vector = zeros(V,1);
+    % loop over all possible ROI pairs
+    for jj = 2:R
+        ii = 1:jj-1;
+        indv = ii + sum(1:jj-2);
+        % Phase difference
+        RP = bsxfun(@minus, theta(:, jj), a_sig_splice(:, ii));
+        srp = sin(RP);
+        pli_vector(indv) = abs(sum(sign(srp),1))/N;
+    end
+
+    % Attempt at converting from a vector to a pli matrix
+    pli_temp = zeros(R, R);
+    pli_temp(ind) = pli_vector(:);
+    pli_temp = pli_temp + pli_temp';
+end
+
+function [pli_corr] = surrogate_analysis(theta, ind, V, pli_temp)
+%% SURROGATE ANALYSIS this is a better way of correcting pli
```

Comment:
why is it a better way?

Reply:
this is a better way of correcting pli than previously found in Jason code since we are fixing the pli using the pli definition instead of the wpli. Mixing pli and wpli is not standard and I'm really not sure what it means.

Added more info in the code